### PR TITLE
Call action "searchContent" with array parameter and with metadata_fi…

### PR DESCRIPTION
**Pull request by ksuess**
…elds array.

Fix https://github.com/plone/volto/issues/2937

```
      searchContent(
        '/',
        {
          portal_type: ['News Item', 'Event'],
          metadata_fields: ['subject', 'modified', 'someotherinfo'],
        },
        'newsandevents',
      ),
```

### CODE REVIEW

**Comment by ksuess** [comment]
See also https://github.com/plone/plone.restapi/pull/1295#issuecomment-998024201

**Comment by sneridagh** [comment]
@ksuess yes... I've been hit by this in plone/plone.restapi#1295 too. 
Problem is that we differed in this long time ago in the expanders... and we built the addExpandersPath thinking on it :(

**Comment by sneridagh** [comment]
So, are tou telling me the search lists are not working since we put in place the AddExpanders thinggy?

**Comment by sneridagh** [comment]
@ksuess I fixed the `addExpandersToPath` helper and added further tests. Now it should work as expected.

Please take a look, let me know what do you think!

**Review by tiberiuichim — APPROVED** [review]

File: src/middleware/api.js [hunk 1]
```diff
@@ -31,24 +38,38 @@ export function addExpandersToPath(path, type) {
...
     encode: false,
   });
-  if (!stringifiedQuery) {
-    return pathPart;
-  }
 
-  return `${pathPart}?${stringifiedQuery}`;
+  if (stringifiedQuery && stringifiedExpand) {
+    return `${url}?${stringifiedExpand}&${stringifiedQuery}`;
+  } else if (!stringifiedQuery && stringifiedExpand) {
```
Comment (nileshgulia1) [inline src/middleware/api.js:66]: Nice. We were indeed missing these conditions which defer queries and expanders working together.

File: packages/core-sandbox/src/components/Views/NewsAndEvents.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    dispatch(
+      searchContent(
+        '/',
+        {
+          portal_type: ['News Item', 'Event'],
+          metadata_fields: ['subject', 'modified', 'someotherinfo'],
+
+          // That's OK:
+          // portal_type: ['News Item'],
+          // metadata_fields: ['subject'],
```
Comment (ksuess) [inline packages/core-sandbox/src/components/Views/NewsAndEvents.jsx:21]: portal_type: ['News Item', 'Event'],
          metadata_fields: ['subject', 'modified', 'someotherinfo'],

**Review by ksuess — COMMENTED** [review]
LGTM

**Pull request merged**